geoportailv3/static/js/search/searchdirective.js

```diff
@@ -197,17 +198,29 @@ app.SearchDirectiveController =
   var POIBloodhoundEngine = this.createAndInitPOIBloodhound_(
       ngeoCreateGeoJSONBloodhound, searchServiceUrl);
 
-  /** @type {Bloodhound} */
-  var LayerBloodhoundEngine = this.createAndInitLayerBloodhound_();
-
-  /** @type {Bloodhound} */
-  var BackgroundLayerBloodhoundEngine = this.createAndInitLayerBloodhound_();
+  /** @type {Fuse} */
+  var LayerFuseEngine =
+      new Fuse(this.layers_, {
+        keys: ['translatedName'],
```

Comment:
You may want to try using the `getFn` option:

```js
var layerFuseEngine = new Fuse(this.layers_, {
  keys: [''],
  getFn: function(item, keys) {
    var suggestion = /** @type {app.BackgroundLayerSuggestion} */ (item);
    return item.translatedName;
  },
  // …
});
```

This would require adding `getFn` to the externs file.

But, looking at your code, there are many variables that should typed but are not. The suggestion variables for example.